test/Microsoft.ML.Tests/Scenarios/Api/AspirationalExamples.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.ML.Tests.Scenarios.Api
+{
+    public class AspirationalExamples
+    {
+        public class IrisPrediction
+        {
+            public string PredictedLabel;
+        }
+
+        public class IrisExample
+        {
+            public float SepalWidth { get; set; }
+            public float SepalLength { get; set; }
+            public float PetalWidth { get; set; }
+            public float PetalLength { get; set; }
+        }
+
+        public void FirstExperienceWithML()
+        {
+            // This is the 'getting started with ML' example, how we see it in our new API.
+            // It currently doesn't compile, let alone work, but we still can discuss and improve the syntax.
+
+            // Load the data into the system.
+            string dataPath = "iris-data.txt";
+            var data = TextReader.FitAndRead(env, dataPath, row => (
```

Comment:
Do we need to be concerned about asynchronous I/O?  Should methods that are I/O bound return `Task<T>`?

Reply:
I think they should have an async analogy, but async-only - I'm not sure

---
In reply to: [208417516](https://github.com/dotnet/machinelearning/pull/639#discussion_r208417516) [](ancestors = 208417516)

Reply:
I sort of remember the discussion, but why do we need to expose FitAndRead to users here? Do we also have a Read case without Fit? Can we simply have something like env.TextReader.Read(datapath, row => ...)?

Reply:
Our existing TextLoader is trainable: it scans the schema of the file and determines number of features under certain conditions.
So, we are essentially calling:
```c#
TextReader.CreateEstimator(row=>...).Fit(dataPath).Read(dataPath)
```
(and discarding the estimator and the reader in the process.

I suppose if we want to discard the reader anyway, we might just as well rename `FitAndRead` to `Read`. But technically, it would still call the above code, so I was reluctant to hide what we are actually doing.

---
In reply to: [208640305](https://github.com/dotnet/machinelearning/pull/639#discussion_r208640305) [](ancestors = 208640305)